tariff/tariffs.go

```diff
@@ -25,6 +25,55 @@ func NewTariffs(currency currency.Unit, grid, feedin, planner api.Tariff) *Tarif
 	}
 }
 
+func currentPrice(t api.Tariff) (float64, error) {
+	if t != nil {
+		if rr, err := t.Rates(); err == nil {
+			if r, err := rr.Current(time.Now()); err == nil {
+				return r.Price, nil
+			}
+		}
+	}
+	return 0, api.ErrNotAvailable
+}
+
+// CurrentGridPrice returns the current feedin price.
+func (t *Tariffs) CurrentGridPrice() (float64, error) {
+	return currentPrice(t.Grid)
+}
+
+// CurrentFeedInPrice returns the current feedin price.
+func (t *Tariffs) CurrentFeedInPrice() (float64, error) {
+	return currentPrice(t.FeedIn)
+}
+
+// CurrentEffectivePrice calculates the real energy price based on self-produced and grid-imported energy.
+func (t *Tariffs) CurrentEffectivePrice(greenShare float64) (float64, error) {
+	if grid, err := t.CurrentGridPrice(); err == nil {
+		feedin, err := t.CurrentFeedInPrice()
+		if err != nil {
+			feedin = 0
+		}
+		return grid*(1-greenShare) + feedin*greenShare, nil
+	}
+	return 0, api.ErrNotAvailable
+}
+
+// CurrentCo2 determines the grids co2 emission.
+func (t *Tariffs) CurrentCo2() (float64, error) {
```

Comment:
Strictly speaking, planner is not tied to grid co2. But I guess we can wave that.

Reply:
Ja, bislang sind unsere Plannerkonfigurationen immer CO2 Datenquellen. Wir könnten auch überlegen `planner` in `co2` umzubenennen und ggf. eine explizite Auswahl zu erlauben ob nach Preis oder nach CO2 optimiert werden soll (bspw. für Tibber User die sich aber auch für CO2 Daten interessieren). Für jetzt würd ich das erstmal so belassen.